runtime/jcl/common/reflecthelp.c

```diff
@@ -837,7 +837,16 @@ createField(struct J9VMThread *vmThread, jfieldID fieldID)
 			|| J9ROMCLASS_IS_RECORD(j9FieldID->declaringClass->romClass)
 			|| J9ROMCLASS_IS_HIDDEN(j9FieldID->declaringClass->romClass)
 		) {
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+			U_32 flags = 0x10; /* java.lang.Field:TRUST_FINAL */
+			if (J9ROMFIELD_IS_NULL_RESTRICTED(j9FieldID->field)) {
+				flags |= 0x20; /* java.lang.Field:NULL_RESTRICTED */
```

Comment:
The code here is inside the conditions from line 835 to 839:
```
if (J9_ARE_ALL_BITS_SET(j9FieldID->field->modifiers, J9AccFinal)) {
		if (J9_ARE_ALL_BITS_SET(j9FieldID->field->modifiers, J9AccStatic)
			|| J9ROMCLASS_IS_RECORD(j9FieldID->declaringClass->romClass)
			|| J9ROMCLASS_IS_HIDDEN(j9FieldID->declaringClass->romClass)
		) {
```
Setting flag `NULL_RESTRICTED` should not be inside these checks.